script/ddgc_get_prev_contributors.pl

```diff
@@ -0,0 +1,159 @@
+my $gh = Net::GitHub->new(access_token => $token);
+
+my @team  = $gh->org->team_members("52625");
+    while($gh->org->has_next_page){
+        push(@team, $gh->org->next_page)
+    }
+my $ddgteam;
+map{ $ddgteam->{$_->{login} }  = 1 } @team;
+
+my $today = localtime;
+# get last days worth of issues
+my $since = $today - (500 * ONE_DAY);
+my $ias;
+
+MAIN :{
+    foreach my $repo (@repos){
+        my $line = 1;
+        
+        warn "Getting all issues since ". $since->datetime;
+
+        my @issues = $gh->issue->repos_issues('duckduckgo', $repo, {
+                state => 'all',
+                since => $since->datetime
+            }
+        );
+
+        while($gh->issue->has_next_page){
+            push(@issues, $gh->issue->next_page)
+        }
+
+        print "Starting $repo\n";
+        my $progress = Term::ProgressBar->new(scalar @issues) unless $d->is_live;
+		
+        # add all the data we care about to an array
+		for my $issue (@issues){
+            $progress->update($line) unless $d->is_live;
+            $line++;
+
+            next if exists $ddgteam->{$issue->{user}->{login}};
+
+            my $state = $issue->{state};
+            next if $state eq 'open';
+
+            $state = $gh->pull_request->is_merged('duckduckgo',$repo, $issue->{number})? 'merged' : $state;
+
+            next if $state ne 'merged';
+
+            # get the IA name from the link in the first comment
+			# Update this later for whatever format we decide on
+			my $name_from_link = '';
+            ($name_from_link) = $issue->{'body'} =~ /https?:\/\/duck\.co\/ia\/view\/(\w+)/i;
+
+            next unless $name_from_link;
+            
+            my $ia = $d->rs('InstantAnswer')->search( {
+                    -or => [
+                        id => $name_from_link,
+                        meta_id => $name_from_link,
+                    ]
+            } )->hri->one_row;
+        
+            next unless $ia;
+
+            my $dev_json;
+            if(exists $ias->{$ia->{id}}){
+                $dev_json = add_developer($ias->{$ia->{id}}, $issue->{user}->{login}, $ia);
+            }else{
+                $dev_json = add_developer($ia->{developer}, $issue->{user}->{login}, $ia);
+            }
+
+            next unless $dev_json;
+
+            $ias->{$ia->{id}} = $dev_json;
+        }
+
+    }
+    
+    foreach my $id (keys $ias){
+        sprintf("update instant_answer set developer = '%s' where id = '%s;'\n",
+            $ias->{$id}, 
+            $id
+        ) >> io("devs.sql")->utf8;
+    }
+}
+
+sub add_developer {
+    my ($dev_json, $author, $ia) = @_;
+    # don't add duplicates
+    return unless $dev_json;
+    return if $dev_json =~ /$author/ig;
+
+    my $user = $d->rs('User')->find_by_github_login($author);
+    my $data;
+
+    try{
+        if($user){
+            my $ddgc_name = $user->username;
+            # Give edit permissions to the contributor
+            $ia->add_to_users($user);
```

Comment:
@jdorweiler please check that the user doesn't have edit permissions already or it will error.
<pre> $ia->add_to_users($user) unless $ia->users->find($user->id) </pre>

You might also want to check whether the user is an admin, since in that case you wouldn't need to give edit permissions.

Also make sure it doesn't error on this line saying $ia is unblessed (aka don't make my same mistakes :smile: )